src/test/java/teammates/test/cases/BaseTestCaseWithDatastoreAccess.java

```diff
@@ -187,6 +187,8 @@ private void equalizeIrrelevantData(AccountAttributes expected, AccountAttribute
                 expected.studentProfile.googleId = actual.googleId;
             }
             expected.studentProfile.modifiedDate = actual.studentProfile.modifiedDate;
+            // data is expected to be sanitized
+            expected.sanitizeForSaving();
```

Comment:
More like, if the data in `expected` is not in the expected format (e.g. sanitized when it should not be or vice versa), we can change the `expected` data.
Adding `expected.sanitizeForSaving()` here seems to add unpredictability to the test behaviour, like you said. `equalizeIrrelevantData` does what it's named after -- equalization to account for unpredictable stuff like date of generation (in many entity types) or unique ID (e.g in feedback questions) or registration key (e.g in students), but sanitized-ness (I wonder if such term exists?) does not seem to fall to that category.